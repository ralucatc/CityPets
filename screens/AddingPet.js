import React, { useState } from 'react'
import * as Yup from 'yup'
import firebase from '../database/firebase'
import { StyleSheet, Image, ScrollView, View, Alert } from 'react-native'
import {
	FormField as Field,
	Form,
    FormImagePicker as ImagePicker,
	FormPicker as Picker,
	SubmitButton as Submit,
} from '../components/forms'
//import database from '@react-native-firebase/database'
//import styles from '../config/styles'


const validationSchema = Yup.object().shape({
	images: Yup.array().min(1, 'Please select at least one image'),
	name: Yup.string().required().min(1).label('Pet name'),
	age: Yup.number().required().min(1).max(10000).label('Age'),
	//category: Yup.string().required().nullable().label('Category'),
	description: Yup.string().label('Description')
});

const categories = [
    { label: "Cats", value: 1},
    { label: "Dogs", value: 2},
    { label: "Birds", value: 3},
    { label: "Others", value: 4},
];

const fields = [
    { label: "Adopt", value: 1},
    { label: "Lost", value: 2},
    { label: "Shelter", value: 3},
];


const submitForm = (values) => {
       var firebaseRef = firebase.database().ref();
       firebaseRef.push().set(values);
       Alert.alert("Your pet was added!");
   }

 export default function AddingPet(){
    return (
        <View style={styles.container}>
            <ScrollView style={styles.scrollView}>
                
                    <Image style={styles.logo} source={require('../assets/form.png')} />
            <Form
            initialValues={{
                name: "",
                age: "",
                description: "",
                fields: null,
                category: null,
                images: []
            }}

            onSubmit={(Values) => submitForm(Values)}
            validationSchema={validationSchema}>

            
            <ImagePicker name='images' />
            <Field maxLength={255} name="name" placeholder="Pet name" />
            <Field
            keyboardType="numeric"
            maxLength={8}
            name="age"
            placeholder="Age"/>
            <Picker
            items={categories}
            name="category"
            placeholder="Category"
            />
          <Picker
            items={fields}
            name="fields"
            placeholder="Fields"
            />
            <Field
            maxLength={255}
            multiline
            name="description"
            numberOfLines={3}
            placeholder="Description"
            />
            <Submit title = "Post" />
            </Form>
            </ScrollView>
            </View>
    )
}


const styles = StyleSheet.create({
    container: {
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        height: '100%'
    },
    logo: {
      width: 400,
      height: 300,
      resizeMode: 'contain',
      justifyContent: 'flex-end',
      alignItems: 'center',
    },
    scrollView: {
        marginHorizontal: 5,
      },
  });



