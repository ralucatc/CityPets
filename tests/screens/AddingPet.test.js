import React from 'react';
import renderer from 'react-test-renderer';
import AddingPet from '../../screens/AddingPet';

describe('<AddingPet />', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<AddingPet />).toJSON();
        expect(tree).toMatchSnapshot();
      });
  });