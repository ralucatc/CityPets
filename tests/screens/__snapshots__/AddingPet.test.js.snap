// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddingPet /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
    }
  }
>
  <RCTScrollView
    style={
      Object {
        "marginHorizontal": 5,
      }
    }
  >
    <View>
      <Image
        source={1}
        style={
          Object {
            "alignItems": "center",
            "height": 300,
            "justifyContent": "flex-end",
            "resizeMode": "contain",
            "width": 400,
          }
        }
      />
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <RCTScrollView
          horizontal={true}
          onContentSizeChange={[Function]}
        >
          <View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#f8f4f4",
                  "borderRadius": 15,
                  "height": 100,
                  "justifyContent": "center",
                  "overflow": "hidden",
                  "width": 100,
                }
              }
            >
              <Text />
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#f8f4f4",
            "borderRadius": 25,
            "flexDirection": "row",
            "marginVertical": 10,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          maxLength={255}
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Pet name"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#0c0c0c",
              "fontFamily": "Avenir",
              "fontSize": 18,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#f8f4f4",
            "borderRadius": 25,
            "flexDirection": "row",
            "marginVertical": 10,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          keyboardType="numeric"
          maxLength={8}
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Age"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#0c0c0c",
              "fontFamily": "Avenir",
              "fontSize": 18,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "#f8f4f4",
              "borderRadius": 25,
              "flexDirection": "row",
              "marginVertical": 10,
              "padding": 15,
            },
            Object {
              "width": "100%",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#0c0c0c",
                "fontFamily": "Avenir",
                "fontSize": 18,
              },
              Object {
                "color": "#6e6969",
                "flex": 1,
              },
            ]
          }
        >
          Category
        </Text>
        <Text />
      </View>
      <Modal
        animationType="slide"
        hardwareAccelerated={false}
        visible={false}
      >
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
              "paddingTop": 44,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Close
              </Text>
            </View>
          </View>
          <RCTScrollView
            data={
              Array [
                Object {
                  "label": "Cats",
                  "value": 1,
                },
                Object {
                  "label": "Dogs",
                  "value": 2,
                },
                Object {
                  "label": "Birds",
                  "value": 3,
                },
                Object {
                  "label": "Others",
                  "value": 4,
                },
              ]
            }
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <View>
              <View
                onLayout={[Function]}
                style={null}
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#0c0c0c",
                          "fontFamily": "Avenir",
                          "fontSize": 18,
                        },
                        Object {
                          "padding": 20,
                        },
                      ]
                    }
                  >
                    Cats
                  </Text>
                </View>
              </View>
              <View
                onLayout={[Function]}
                style={null}
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#0c0c0c",
                          "fontFamily": "Avenir",
                          "fontSize": 18,
                        },
                        Object {
                          "padding": 20,
                        },
                      ]
                    }
                  >
                    Dogs
                  </Text>
                </View>
              </View>
              <View
                onLayout={[Function]}
                style={null}
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#0c0c0c",
                          "fontFamily": "Avenir",
                          "fontSize": 18,
                        },
                        Object {
                          "padding": 20,
                        },
                      ]
                    }
                  >
                    Birds
                  </Text>
                </View>
              </View>
              <View
                onLayout={[Function]}
                style={null}
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#0c0c0c",
                          "fontFamily": "Avenir",
                          "fontSize": 18,
                        },
                        Object {
                          "padding": 20,
                        },
                      ]
                    }
                  >
                    Others
                  </Text>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </RCTSafeAreaView>
      </Modal>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "#f8f4f4",
              "borderRadius": 25,
              "flexDirection": "row",
              "marginVertical": 10,
              "padding": 15,
            },
            Object {
              "width": "100%",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#0c0c0c",
                "fontFamily": "Avenir",
                "fontSize": 18,
              },
              Object {
                "color": "#6e6969",
                "flex": 1,
              },
            ]
          }
        >
          Fields
        </Text>
        <Text />
      </View>
      <Modal
        animationType="slide"
        hardwareAccelerated={false}
        visible={false}
      >
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
              "paddingTop": 44,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Close
              </Text>
            </View>
          </View>
          <RCTScrollView
            data={
              Array [
                Object {
                  "label": "Adopt",
                  "value": 1,
                },
                Object {
                  "label": "Lost",
                  "value": 2,
                },
                Object {
                  "label": "Shelter",
                  "value": 3,
                },
              ]
            }
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <View>
              <View
                onLayout={[Function]}
                style={null}
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#0c0c0c",
                          "fontFamily": "Avenir",
                          "fontSize": 18,
                        },
                        Object {
                          "padding": 20,
                        },
                      ]
                    }
                  >
                    Adopt
                  </Text>
                </View>
              </View>
              <View
                onLayout={[Function]}
                style={null}
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#0c0c0c",
                          "fontFamily": "Avenir",
                          "fontSize": 18,
                        },
                        Object {
                          "padding": 20,
                        },
                      ]
                    }
                  >
                    Lost
                  </Text>
                </View>
              </View>
              <View
                onLayout={[Function]}
                style={null}
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#0c0c0c",
                          "fontFamily": "Avenir",
                          "fontSize": 18,
                        },
                        Object {
                          "padding": 20,
                        },
                      ]
                    }
                  >
                    Shelter
                  </Text>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </RCTSafeAreaView>
      </Modal>
      <View
        style={
          Object {
            "backgroundColor": "#f8f4f4",
            "borderRadius": 25,
            "flexDirection": "row",
            "marginVertical": 10,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          maxLength={255}
          multiline={true}
          numberOfLines={3}
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Description"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#0c0c0c",
              "fontFamily": "Avenir",
              "fontSize": 18,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#57419d",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 18,
              "fontWeight": "bold",
              "textTransform": "uppercase",
            }
          }
        >
          Post
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
